mscore/pianoroll/pianoroll.cpp

```diff
@@ -773,18 +789,22 @@ void PianorollEditor::posChanged(POS p, unsigned tick)
 void PianorollEditor::onTimeChanged(int val)
       {
       QList<PianoItem*> items = pianoView->getSelectedItems();
-      if (items.size() != 1)
-            return;
-      PianoItem* item = items[0];
-      Note* note       = item->note();
-      NoteEvent* event = item->getTweakNoteEvent();
-      if (!event || event->ontime() == val)
+      if (!items.size())
             return;
 
-      NoteEvent ne = *event;
-      ne.setOntime(val);
       _score->startCmd();
-      _score->undo(new ChangeNoteEvent(note, event, ne));
+      for (int i = 0; i < items.size(); i++) {
```

Comment:
Would you mind extracting this code to a separate method? I see the code duplication in `PianorollEditor::tickLenChanged` method.

Reply:
Is it really necessary? I see function differences such as `event->ontime()` and `event->len()`. Even if I use a template function, it's hard to make these member functions a kind of variables.